sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -851,7 +851,7 @@ private[hive] object HiveClientImpl {
       hiveTable.setFields(schema.asJava)
     }
     hiveTable.setPartCols(partCols.asJava)
-    conf.foreach(c => hiveTable.setOwner(c.getUser))
+    conf.foreach(c => hiveTable.setOwner(SessionState.get().getAuthenticator().getUserName()))
```

Comment:
Thank you for review, @vanzin !
Which `object` do you mean for `.foreach` here?
For `conf`, we already use that.
For `SessionState.get()`, it's not `Option`.